dask/array/gufunc.py

```diff
@@ -351,8 +351,11 @@ def apply_gufunc(func, signature, *args, **kwargs):
         shape = arg.shape
         iax = tuple(a if a < 0 else a - len(shape) for a in iax)
         tidc = tuple(i for i in range(-len(shape) + 0, 0) if i not in iax) + iax
-
-        transposed_arg = arg.transpose(tidc)
+        if tidc == tuple(range(-arg.ndim, 0)):
+            # transpose not necessary
```

Comment:
For my understanding, this means that `arg == arg.transpose(tidc)`?

Reply:
correct

Reply:
We might also make this fix in the transpose function itself.  That would also be easier to test.

```python
assert x.transpose((1, 2, 3)) is x
```